utils/prepare_results_in_kaldi_format.py

```diff
@@ -24,3 +24,72 @@ def combine_asr_data(input_dirs, output_dir):
+
+def check_file(ori_file, anon_file):
+    skip=True
+    if 'wav.scp' in str(anon_file):
+        for line in open(anon_file):
+            if 'anon_wav' in line.strip():
+                skip += os.path.exists(line.strip().split(' ')[-1])
+            else:
+                return False
+    else:
+        with open(ori_file, 'rb') as f1, open(anon_file, 'rb') as f2:
+            content1 = f1.read()
+            content2 = f2.read()
+            if content1 == content2:
+                skip=True
+    return skip
+
+
+def check_kaldi_formart_data(config): 
+    dataset_dict = get_datasets(config)
+    output_path = config['data_dir']
+    suffix = config['anon_data_suffix']
+
+    for dataset, orig_dataset_path in dataset_dict.items():
+        # do the transformation for original and anonymized versions of each dataset
+        # if it is the original data, the information is simply copied from an external source of the data
+        # if it is the anonymized data, some information (e.g. the wav.scp) is generated to match the information of the anonymized data
+        
+        out_data_split = output_path / f'{dataset}{suffix}'
+        #out_data_split.mkdir(exist_ok=True, parents=True)
+        if not os.path.exists(out_data_split):
+            print(f"Directory {out_data_split} does not exist. Please prepare your anoymized audio in a correct formart.")
+            exit()
+        
+        # these files are never changed during anonymization and always copied from the original
+        copy_files = ['spk2utt', 'text', 'utt2spk','spk2gender','utt2dur', "wav.scp"]
+
+        if 'trials' in dataset:
+            # the trials file only exists for the trials subsets
+            copy_files += ['trials']
+        elif 'enrolls' in dataset:
+            # the enrolls file only exists for the enrolls subsets
+            copy_files += ['enrolls']
+
+        em_copy_files = ['spk2fold', 'spk2gender','spk2utt',  'text', 'text_no_norm','utt2emo','utt2spk', "wav.scp"]
+
+        if "IEMOCAP" in dataset:
+            copy_files = em_copy_files
+
+        for file in copy_files:
+            ori_file = orig_dataset_path / file
+            anon_file = out_data_split / file
+            skip = False
+            
+            # Check if  anonymized file already exists and check
+            if os.path.exists(anon_file):
+                skip = check_file(ori_file, anon_file)
+            
+            # Create anonymized file if not exits or not correct
+            if not skip:
+                copy(ori_file, anon_file)
+                if file=='wav.scp':
+                    fp = open(anon_file,'w')
+                    for line in open(ori_file):
+                        temp = line.strip().split(' ')
+                        token = temp[0]
+                        audio_path = out_data_split / 'anon_wav' / token
+                        fp.write("%s %s.wav\n"%(token, audio_path))
+                    fp.close()
```

Comment:
This function has a "IEMOCAP" check, making it not that generalizable to other datasets/tasks. 
A better way to check would be to unsure all file (not dirs) that are in the non-anonymized directory (e.g.: `IEMOCAP_dev` are present in `IEMOCAP_DEV_$SUFFIX`).
The function name is `check_kaldi_formart_data` but it appears to to more than that, (Create anonymized file if not exits or not correct). I propose to split the function in two, `check_kaldi_formart_data` and `create_kaldi_wavscp(config)` (should only work for anon data dirs).
The `'anon_wav'` line 93 is also hard coded, while being optional in anon_xxx.yaml file.

Reply:
I think the `Create anonymized file if not exits or not correct` part should be removed, it should be created by the pipeline.
I don't think having the 'anon_wav' and the '.wav' hardcoded is good.